build-conventions/src/main/java/org/elasticsearch/gradle/internal/conventions/info/ParallelDetector.java

```diff
@@ -90,4 +94,15 @@ private static boolean isMac(ProviderFactory providers) {
         return providers.systemProperty("os.name").getOrElse("").startsWith("Mac");
     }
 
+    private static boolean isMontereyOrNewer(ProviderFactory providers) {
+        String rawVersion = providers.systemProperty("os.version").getOrElse("").trim();
+        if (rawVersion.isEmpty()) {
+            Logging.getLogger(ParallelDetector.class).warn("Failed to validate MacOs version.");
```

Comment:
Can we refactor the logger out into a static field? It's only used once here but it'll help us not duplicating this if we add more logging to the class down the road.

Reply:
Sure! I just added a final static variable called LOGGER with that. Let me know if it looks good to you!